### PR TITLE
Fix some of the issues raised by analyzers

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharp.java [hunk 1]
```diff
@@ -36,4 +36,8 @@ public CSharp(Settings settings) {
     return settings.getStringArray(CSharpPlugin.FILE_SUFFIXES_KEY);
   }
 
+  @Override
+  public boolean equals(Object o) {
```
Comment (SonarTech) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharp.java]: MINOR This class overrides "equals()" and should therefore also override "hashCode()". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1206)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharp.java (hunk 1 above)
Comment (henryju) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharp.java]: Why have you added this? Seems totally useless to override a method just to call super on it...

Comment (Evangelink) [inline sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharp.java]: Unfinished job sorry, wanted to fix the warning and didn't finished the implementation before getting disrupted.

**Pull request merged**